test/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,122 @@
+using Random, Turing, Test
+using Clustering
+
+dir = splitdir(splitdir(pathof(Turing))[1])[1]
+include(dir*"/test/test_utils/AllUtils.jl")
+
+
+@turing_testset "gibbs conditionals" begin
+    let N = 1000,
+        (α_0, θ_0) = (2.0, inv(3.0)),
+        λ_true = rand(Gamma(α_0, θ_0)),
+        σ_true = sqrt(1 / λ_true),
+        m_true = rand(Normal(0, σ_true)),
+        x = rand(Normal(m_true, σ_true), N)
+        
+        @model function inverse_gdemo(x)
+            λ ~ Gamma(α_0, θ_0)
+            m ~ Normal(0, sqrt(1 / λ))
+            @. x ~ $(Normal(m, sqrt(1 / λ)))
+        end
+
+        function gdemo_statistics(x)
+            # The conditionals and posterior can be formulated in terms of the following statistics:
+            n = length(x) # number of samples
+            x_bar = mean(x) # sample mean
+            s2 = var(x; mean=x_bar, corrected=false) # sample variance
+            return n, x_bar, s2
+        end
+
+        function cond_m(c)
+            n, x_bar, s2 = gdemo_statistics(x)
+            m_n = N * x_bar / (n + 1)
+            λ_n = c.λ * (n + 1)
+            σ_n = sqrt(1 / λ_n)
```

Comment:
Wouldn't it be easier to precompute `m_n` and a scaled variance outside of the function instead of closing over `x` and `N`? It is not clear if the compiler eliminates the redundant computations (it's also not possible for the square root).